### ISSUE TITLE
Define the elements / fields / sections of the GUI

### ISSUE THREAD

**Issue by alexitsios** [alexitsios/Calamity#105]

**Comment by teberzin** [comment]
@alexitsios I think this task is completed.

**Comment by alexitsios** [comment]
Not all elements of the GUI have been defined, but only the inventory from what I see.

**Comment by alexitsios** [comment]
The main areas are:

Main Menu
Options/settings menu
Game Menu (incl. inventory, files, map, etc)

**Comment by alexitsios** [comment]
if you need help from the programmers, I suggest you ping them here @teberzin and ask for clarifications.

**Comment by teberzin** [comment]
@seekeroftheball Hey Jordan, 

I need your help with the GUI. I've already define the elements of the inventory screen. We will also have the settings screen. Except quit option mostly it will contain the accessibility settings. I'm not familiar with these settings and what settings we can have techincally. 

So in short can you help me to define elements on the settings screen? If you need anything from my end you can let me know.

**Comment by seekeroftheball** [comment]
@teberzin Absolutely, and this is something I've been considering already. Accessibility is a broad subject and from a technical standpoint there's not really a limitation to what we can achieve. Since we're shooting for a commercial release, accessibility features can also help with our marketing, as some marketplaces actually host regular showcases for games with strong accessibility features.

Ultimately though, the features we include are a design choice. I'll link to a great website that shows a list of the most common accessibility features. I can offer a few suggestions on what basics we should include, but also a good resource is to look up screenshots of settings menus from games you enjoy.

https://gameaccessibilityguidelines.com

A few basics to get started:
 - Button / controls remapping
 - Look inversion
 - Screen calibration
	- Resolution settings for most common monitor resolutions.
	- Brightness, contrast, and gamma adjustments.
 - Color blind mode / UI color customization
 - Audio adjustment settings
	- Volume sliders
	- Hardware output options
 - Field of view
 - Difficulty options
 
Some settings like difficulty will effect game design so they should be considered through that lens; if we'd like to include them or not.

I hope that helps. Let me know any questions or if I can offer more suggestions.

**Comment by alexitsios** [comment]
@ChanceADev can you shed some light here? We want primarily to define the elements and section.

etc.

Main menu
-Start
-Options
-About
...
-Exit

Game Menu
-Files
-Inventory
...

Options
-Master Volume
-SFX
...

We need this to have our GUI artist start working on it. Since you have experience with UI programming, can you help us  define the elements, sections, etc. ?

**Comment by alexitsios** [comment]
@chaserimmer @mashewmake please provide any feedback / help you might have here. At this point we mainly want to focus on what the GUI will have.

**Comment by teberzin** [comment]
@alexitsios We have the Game Menu defined already: https://github.com/alexitsios/Calamity/wiki/Gameplay#game-menu
I think GUI artist can start working for it meanwhile I will define the Main Menu and Options. 
If GUI artist have any question regarding the Game Menu they can reach me.

**Comment by ChanceADev** [comment]
> We need this to have our GUI artist start working on it. Since you have experience with UI programming, can you help us define the elements, sections, etc. ?

I may be misunderstanding. But depending on how the UI is being represented, if the art is appropriately separated into small reusable pieces, 9-sliced, etc, there should be no problem in engine with adding interactivity or rescaling for different aspect ratios or any other UI programming task.

**Comment by teberzin** [comment]
@alexitsios Done.
